cpp/src/arrow/testing/generator.h

```diff
@@ -23,6 +23,7 @@
 #include <vector>
 
 #include "arrow/array/array_base.h"
+#include "arrow/builder.h"
```

Comment:
This is a very large inclusion, can we avoid it here? Perhaps only include `arrow/array/builder_primitive.h`?

Reply:
Or you can perhaps just rely on `TypedBufferBuilder<T>` below.